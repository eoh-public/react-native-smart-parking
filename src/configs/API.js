import Config from 'react-native-config';

const API_ROOT = Config.API_ROOT;

const API = {
  AUTH: {
    LOGIN_PHONE: API_ROOT + '/accounts/login/',
    LOGIN_FACEBOOK: API_ROOT + '/accounts/login/facebook/',
    REGISTER_PHONE: API_ROOT + '/accounts/register/phone/',
    RESEND_OTP: API_ROOT + '/accounts/register/resend_otp/',
    VERIFY_OTP: API_ROOT + '/accounts/register/verify_otp/',
    LOGIN_SOCIAL_FB: API_ROOT + '/accounts/login/facebook/',
    LOGIN_SOCIAL_GG: API_ROOT + '/accounts/login/google/',
    LOGIN_SOCIAL_AP: API_ROOT + '/accounts/login/apple/',
    FORGOT_PASSWORD: API_ROOT + '/accounts/forgot_password/',
    FORGOT_PASSWORD_VERIFY_OTP:
      API_ROOT + '/accounts/forgot_password/verify_otp/',
    RESET_PASSWORD: API_ROOT + '/accounts/forgot_password/set_password/',
  },
  ACCOUNTS: {
    FEATURES: API_ROOT + '/accounts/features/',
    ADD_CARD: API_ROOT + '/billing/payments/stripe/add_card/',
    REMOVE_CARD: (id) =>
      API_ROOT + `/billing/payments/stripe/remove_cards/${id}/`,
    CHANGE_DEFAULT_CARD: API_ROOT + '/billing/payments/stripe/set_default/',
    LIST_PAYMENT_METHODS: API_ROOT + '/billing/list_payment_methods/',
    CREATE_CARD_TOKEN: 'https://api.stripe.com/v1/tokens',
  },
  UNIT: {
    ADD_USER: API_ROOT + '/property_manager/shared_units/add_by_phone_number/',
    MY_UNITS: API_ROOT + '/property_manager/units/mine/',
    SHARED_UNITS: API_ROOT + '/property_manager/shared_units/',
    UNIT_DETAIL: (id) => API_ROOT + `/property_manager/units/${id}/`,
    UNIT_NEAR_ME: (lat, lon, page) =>
      API_ROOT + `/property_manager/units/near-me/${lat},${lon}/?page=${page}`,
    UNITS_PUBLIC: API_ROOT + '/property_manager/units/public/',
    UNIT_SUMMARY: (id) => API_ROOT + `/property_manager/units/${id}/summary/`,
    UNIT_SUMMARY_DETAIL: (id, id2) =>
      API_ROOT + `/property_manager/units/${id}/summary_detail/${id2}/`,
    FILTER_SHARED_UNITS: (field) =>
      API_ROOT + `/property_manager/shared_units/?ordering=${field}`,
    STAR_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/star/`,
    UNSTAR_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/unstar/`,
    PIN_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/pin/`,
    UNPIN_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/unpin/`,
    MANAGE_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/`,
  },
  SUB_UNIT: {
    REMOVE_SUB_UNIT: (unitId, id) =>
      API_ROOT + `/property_manager/${unitId}/sub_units/${id}/`,
    CREATE_SUB_UNIT: (unitId) =>
      API_ROOT + `/property_manager/${unitId}/sub_units/`,
    MANAGE_SUB_UNIT: (unitId, id) =>
      API_ROOT + `/property_manager/${unitId}/sub_units/${id}/`,
    SENSOR_SCAN: (id) =>
      API_ROOT + `/property_manager/stations/${id}/sensor_scan/`,
  },
  SENSOR: {
    DISPLAY: (id) => API_ROOT + `/property_manager/sensors/${id}/display/`,
    DISPLAY_VALUES: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/display_values/`,
    DISPLAY_VALUES_V2: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/display_values_v2/`,
    DISPLAY_HISTORY: (id, hId) =>
      API_ROOT + `/property_manager/sensors/${id}/display_history/`,
    CAMERA_DISPLAY: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/cameras/`,
    REMOTE_CONTROL_OPTIONS: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/remote_control_options/`,
    QUICK_ACTION: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/quick_action/`,
    CHECK_CONNECTION: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/check_connection/`,
    UPDATE_SENSOR: (unit_id, station_id, id) =>
      API_ROOT +
      `/property_manager/${unit_id}/sub_units/${station_id}/devices/${id}/`,
  },
  CONFIG: {
    DISPLAY_HISTORY: API_ROOT + '/property_manager/configs/display_history/',
  },
  POWER_CONSUME: {
    DISPLAY_HISTORY:
      API_ROOT + '/property_manager/power_consume/display_history/',
  },
  SHARE: {
    UNITS: API_ROOT + '/property_manager/sharing/units/',
    UNIT_PERMISSIONS: (id) =>
      API_ROOT + `/property_manager/sharing/units/${id}/permissions/`,
    UNITS_MEMBERS: (id) =>
      API_ROOT + `/property_manager/sharing/units/${id}/members/`,
    UNITS_MEMBER_DETAIL: (id, shareId) =>
      API_ROOT + `/property_manager/sharing/units/${id}/members/${shareId}/`,
    SHARE: API_ROOT + '/property_manager/sharing/share/',
  },
  EXTERNAL: {
    WEATHER: (location, weatherMapId) =>
      `https://api.openweathermap.org/data/2.5/weather?lat=${location.lat}&lon=${location.lon}` +
      `&appid=${weatherMapId}`,
    UV_INDEX: (location, weatherMapId) =>
      `https://api.openweathermap.org/data/2.5/uvi?appid=${weatherMapId}&lat=${location.lat}&lon=${location.lon}`,
    AQI: (location, weatherMapToken) =>
      `https://api.waqi.info/feed/geo:${location.lat};${location.lon}/?token=${weatherMapToken}`,
    GOOGLE_MAP: {
      AUTO_COMPLETE:
        'https://maps.googleapis.com/maps/api/place/autocomplete/json',
      GET_LAT_LNG_BY_PLACE_ID:
        'https://maps.googleapis.com/maps/api/place/details/json',
    },
  },
  PARKING: {
    NEARBY: API_ROOT + '/smart_parking/parking_areas/near_me/',
    NEAREST:
      API_ROOT + '/smart_parking/parking_areas/nearest_available_parking/',
    SAVED_LIST: API_ROOT + '/smart_parking/parking_areas/saved_list/',
    SAVE: (id) => API_ROOT + `/smart_parking/parking_areas/${id}/save/`,
    UNSAVE: (id) => API_ROOT + `/smart_parking/parking_areas/${id}/unsave/`,
    DETAIL: (id) => API_ROOT + `/smart_parking/parking_areas/${id}/`,
    GET_BOOKING_PRICE: (id) =>
      API_ROOT + `/smart_parking/parking_areas/${id}/get_price/`,
    AVAILABLE_TIME_SLOTS: (id) =>
      API_ROOT + `/smart_parking/parkings/${id}/available-time-slots/`,
    PAYMENT_SUCCESS: (id) => API_ROOT + `/smart_parking/payment_success/${id}/`,
    CHECK_CAR_PARKED: API_ROOT + '/smart_parking/spots/check_car_parked/',
    PARKING_INFO: API_ROOT + '/smart_parking/spots/parking_info/',
  },
  CAR: {
    MY_CARS: API_ROOT + '/smart_parking/cars/',
    CHECK_CARS_INFO: API_ROOT + '/smart_parking/cars/check_information/',
    REMOVE_CAR: (id) => API_ROOT + `/smart_parking/cars/${id}/`,
    UPDATE: (id) => API_ROOT + `/smart_parking/cars/${id}/`,
    UPDATE_DEFAULT: (id) =>
      API_ROOT + `/smart_parking/cars/${id}/update_default/`,
  },
  BOOKING: {
    CREATE: API_ROOT + '/smart_parking/bookings/',
    DETAIL: (id) => API_ROOT + `/smart_parking/bookings/${id}/`,
    ACTIVE_SESSION: API_ROOT + '/smart_parking/bookings/active_session/',
    HISTORY: (page) =>
      API_ROOT + `/smart_parking/bookings/history/?page=${page}`,
    SCAN_TO_BOOK: API_ROOT + '/smart_parking/bookings/scan_to_book/',
    SCAN_TO_CONFIRM: API_ROOT + '/smart_parking/bookings/scan_to_confirm/',
    EXTEND_INFO: (id) =>
      API_ROOT + `/smart_parking/bookings/${id}/extend_info/`,
    EXTEND: (id) => API_ROOT + `/smart_parking/bookings/${id}/extend/`,
    CANCEL: (id) => API_ROOT + `/smart_parking/bookings/${id}/cancel/`,
    STOP: (id) => API_ROOT + `/smart_parking/bookings/${id}/stop/`,
  },
  EMERGENCY: {
    CONTACTS: API_ROOT + '/emergency_button/contacts/',
  },
  BILLING: {
    LIST_PAYMENT_METHODS_BY_COUNTRY: (code) =>
      API_ROOT + `/billing/list-methods/${code}/`,
    PAYMENT: {
      STRIPE: {
        PROCESS: (id) => API_ROOT + `/billing/payments/stripe/process/${id}/`,
        ADD_CARD: API_ROOT + '/billing/payments/stripe/add_card/',
        CREATE_PAYMENT_INTENT: (id) =>
          API_ROOT + `/billing/payments/stripe/sca/create-payment/${id}/`,
        PAYMENT_INTENT_SUCCESS: (intent_id) =>
          API_ROOT +
          `/billing/payments/stripe/sca/payment_success/${intent_id}/`,
      },
    },
    DEFAULT_PAYMENT_METHODS: API_ROOT + '/billing/default_payment_method/',
  },
  NOTIFICATION: {
    REGISTER_SIGNAL_ID: API_ROOT + '/accounts/register-signal-id/',
    LIST_ALL_NOTIFICATIONS: (page, type) =>
      API_ROOT + `/notifications/notifications/?page=${page}&type=${type}`,
    SET_READ: (id) => API_ROOT + `/notifications/notifications/${id}/set_read/`,
    NUMBER: API_ROOT + '/notifications/notifications/number/',
  },
  EMERGENCY_BUTTON: {
    CREATE_CONTACT: API_ROOT + '/emergency_button/contacts/',
    CONTACTS: API_ROOT + '/emergency_button/contacts/',
    REMOVE_CONTACTS: (id) => API_ROOT + `/emergency_button/contacts/${id}/`,
    SEND_ALERT: API_ROOT + '/emergency_button/events/',
    NOTIFY_SMS: (id) => API_ROOT + `/emergency_button/events/${id}/notify_sms/`,
    FASLE_ALARM: (id) =>
      API_ROOT + `/emergency_button/events/${id}/false_alarm/`,
    RESOLVE: (id) => API_ROOT + `/emergency_button/events/${id}/resolve/`,
  },
  IOT: {
    CHIP_MANAGER: {
      WATCH_CONFIGS: API_ROOT + '/chip_manager/watch_configs/',
      PUSHER_AUTH: API_ROOT + '/chip_manager/pusher/auth/',
    },
  },
};

export default API;
