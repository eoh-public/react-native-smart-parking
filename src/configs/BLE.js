export default {
  BLE_REMOTE_SERVICE_UUID: '4fafc201-1fb5-459e-8fcc-c5c9c331914b',
  BLE_REMOTE_CHARACTERISTIC_UUID: 'beb5483e-36e1-4688-b7f5-ea07361b26a8',
};
