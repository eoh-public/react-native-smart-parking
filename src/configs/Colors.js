export const Colors = {
  Primary: '#00979D',
  Text: '#363636',
  Golden: '#b2973f',
  White: '#FFFFFF',
  Red: 'red',
  Green: 'green',
  Gray: '#838383',
  LightGray: '#989898',
  Error: '#F10303',
  DarkLayer: '#00000080',
  LightDarkLayer: '#0000004C',
  LightingDarkLayer: '#00000033',
  LightWhiteLayer: '#F7F7F7D9',
  CoinTagBG: '#F7F7F7',
  Black: '#000000',
  Disabled: '#f8f8f8',
  Border: '#BFBFBF',
  ButtonTag: '#989898',
  WrapGray: '#F5F5F5',
  SwitchButton: '#909090',
  Read: '#F3F3F3',
  Orange: '#F08229',
  BG: '#F0F2F5',
  BGImage: '#F4F8F8',
  Transparent: '#00000000',
  TransparentWhite: '#FFFFFF00',
  BGMapCircle: '#E6F7FF30',
  BGNotRain: '#FEFFE6',
  BorderRain: '#36CFC9',
  Shadow: '#303133',
  ShadownTransparent: 'rgba(48, 49, 51, 0.1)',
  TextTransparent: 'transparent',
  TextGray: 'gray',
  LightOrange: '#FEEFE7',

  //Range Cyan
  Cyan2: '#B5F5EC',
  Cyan2opa30: '#B5F5EC4D',
  Cyan6: '#13C2C2',

  //Range GeekBlue
  GeekBlue3: '#ADC6FF',

  // Range Gray
  Gray4: '#E8E8E8',
  Gray5: '#D9D9D9',
  Gray6: '#BFBFBF',
  Gray7: '#8C8C8C',
  Gray8: '#595959',
  Gray9: '#262626',
  Gray2: '#FAFAFA',
  Gray3: '#F5F5F5',
  Gray11: '#3031331a',
  Gray12: '#DDD',
  Gray13: '#CCC',
  Gray14: '#EEE',
  Gray16: '#333',
  Gray17: '#D3D3D3',
  Gray18: '#9B9A9B',
  Gray19: '#E5E5E5',

  // Range Green
  Green1: '#F6FFED',
  Green3: '#B7EB8F',
  Green6: '#52C41A',
  Green7: '#00979D',
  Green8: '#D9F7BE',
  Green9: '#389E0D',
  Green10: '#237804',

  //Yellow
  Yellow6: '#FADB14',
  Yellow4: '#FFF566',

  //Orange
  Orange3: '#FFD591',
  Orange4: '#FFC069',
  Orange6: '#FA8C16',
  Orange7: '#FAAD14',

  //Red
  Red9: '#A8071A',
  Red8: '#CF1322',
  Red7: '#FA541C',
  Red6: '#F5222D',
  Red5: '#FF4D4F',
  Red4: '#FF7875',
  Red3: '#FFA39E',
  Red2: '#FFCCC7',
  Red1: '#FFF1F0',

  //Purple
  Purple3: '#D3ADF7',
  Purple6: '#722ED1',

  //Pink
  Pink1: '#EFDBFF',

  //Blue
  Blue1: '#F0F5FF',
  Blue2: '#D6E4FF',
  Blue3: '#ADC6FF',
  Blue4: '#85A5FF',
  Blue5: '#597EF7',
  Blue6: '#2F54EB',
  Blue7: '#1D39C4',
  Blue9: '#91D5FF',
  Blue10: '#1890FF',
  Blue11: '#096DD9',
  Blue12: '#0050B3',

  //Range Volcano:
  Volcano3: '#FFBB96',

  //lime
  Lime6: '#A0D911',

  Cyan1: '#E6FFFB',

  LightPrimary: '#EBF7F8',

  BlackTransparent1: 'rgba(0, 0, 0, 0.1)',
  BlackTransparent2: 'rgba(0, 0, 0, 0.2)',
  BlackTransparent3: 'rgba(0, 0, 0, 0.3)',
  BlackTransparent4: 'rgba(0, 0, 0, 0.4)',
  BlackTransparent5: 'rgba(0, 0, 0, 0.5)',
  BlackTransparent6: 'rgba(0, 0, 0, 0.6)',
  BlackTransparent16: 'rgba(0, 0, 0, 0.16)',
  WhiteTransparent8: 'rgba(255, 255, 255, 0.8)',
};
