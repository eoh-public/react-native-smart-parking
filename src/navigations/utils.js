import * as React from 'react';

// NavigationContainer is referred here - Check NavigationStack
export const navigationRef = React.createRef();

export function navigate(name, params) {
  navigationRef.current?.navigate(name, params);
}

export function goBack() {
  navigationRef.current?.goBack();
}

export default {
  navigate,
  goBack,
};
