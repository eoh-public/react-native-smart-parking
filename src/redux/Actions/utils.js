const REQUEST = 'REQUEST';
const SUCCESS = 'SUCCESS';
const FAIL = 'FAIL';
const RESET = 'RESET';

export const createRequestTypes = (base) => ({
  [REQUEST]: `${base}_${REQUEST}`,
  [SUCCESS]: `${base}_${SUCCESS}`,
  [FAIL]: `${base}_${FAIL}`,
  [RESET]: `${base}_${RESET}`,
});
