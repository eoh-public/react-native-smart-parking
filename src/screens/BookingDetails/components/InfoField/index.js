import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';
import { Colors } from '../../../../configs';
import Text from '../../../../commons/Text';
import {t} from 'i18n-js';;

const InfoField = memo(({ value, title, style, body = true, onDirection }) => {
  return (
    <View style={[styles.container, style && style]}>
      <Text type={'Body'} color={Colors.Gray7} style={styles.text}>
        {title}
      </Text>
      <Text
        type={body ? 'Body' : 'H4'}
        color={Colors.Gray9}
        style={styles.text}
        semibold={!body}
      >
        {value}
        {'    '}
        {onDirection && (
          <Text type={'Body'} color={Colors.Primary} onPress={onDirection}>
            {t('directions')}
          </Text>
        )}
      </Text>
    </View>
  );
});

export default InfoField;

const styles = StyleSheet.create({
  text: {
    marginTop: 8,
  },
  container: {
    flex: 1,
  },
});
