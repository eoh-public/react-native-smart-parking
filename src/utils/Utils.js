import { Linking } from 'react-native';
import Toast from 'react-native-toast-message';
import axios from 'axios';
import { Constants } from '../configs';

export const setAxiosDefaultAuthToken = (token) => {
  axios.defaults.headers.common.Accept = 'application/json';
  axios.defaults.headers.common.Authorization = `Token ${token}`;
};

export const setAxiosDefaultLanguage = (language) => {
  axios.defaults.headers.common['Accept-Language'] =
    language || Constants.LANGUAGE.DEFAULT;
};

export const isObjectEmpty = (obj) => {
  return Object.keys(obj).length === 0;
};

export const formatMoney = (number, fixed = 0, currency = 'VND') => {
  return (
    parseInt(number, 10)
      .toFixed(fixed)
      .toString()
      .replace(/(\d)(?=(\d\d\d)+(?!\d))/g, '$1.') + ` ${currency}`
  );
};

export const openMapDirection = (item) => () => {
  // https://developers.google.com/maps/documentation/urls/get-started#universal-cross-platform-syntax
  const url = `https://www.google.com/maps/dir/?api=1&origin=&destination=${item.lat},${item.lng}`;
  Linking.openURL(url);
};

export const ToastBottomHelper = {
  success: (msg) => {
    if (!Toast._ref) {
      return;
    }

    Toast.show({
      type: 'success',
      position: 'bottom',
      text1: msg,
      visibilityTime: 1000,
    });
  },
  error: (msg) => {
    if (!Toast._ref) {
      return;
    }

    Toast.show({
      type: 'error',
      position: 'bottom',
      text1: msg,
      visibilityTime: 1000,
    });
  },
};

export const insertToString = (str, index, value) => {
  return str.substr(0, index) + value + str.substr(index);
};

export const removeFromString = (str, index) => {
  return str.substr(0, index) + str.substr(index + 1);
};

export default {
  isObjectEmpty,
  insertToString,
};
