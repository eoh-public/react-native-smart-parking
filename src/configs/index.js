import { SPConfig, initSPConfig } from './SPConfig';
import API from './API';
import { Colors } from './Colors';
import { Constants } from './Constants';
import Theme from './Theme';
import Images from './Images';
import Styles from './Styles';
import BLE from './BLE';
import Device from './Device';
import { Fonts } from './Fonts';
export {
  API,
  BLE,
  Colors,
  Constants,
  Theme,
  Styles,
  Images,
  Device,
  Fonts,
  SPConfig,
  initSPConfig,
};
