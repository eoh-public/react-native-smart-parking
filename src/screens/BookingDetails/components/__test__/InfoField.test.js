import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { Colors } from '../../../../configs';
import { openMapDirection } from '../../../../utils/Utils';
import InfoField from '../InfoField';

describe('Test InfoField', () => {
  let tree;

  test('render info field item without body', () => {
    act(() => {
      tree = renderer.create(
        <InfoField
          value="value"
          title="title"
          style={{ backgroundColor: Colors.White }}
          onDirection={openMapDirection({
            lat: 0,
            lng: 0,
          })}
        />
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('render info field item with body', () => {
    act(() => {
      tree = renderer.create(
        <InfoField value="value" title="title" body={false} />
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
