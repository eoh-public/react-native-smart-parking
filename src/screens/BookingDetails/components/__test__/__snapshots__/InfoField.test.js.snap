// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InfoField render info field item with body 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#8C8C8C",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginTop": 8,
        },
      ]
    }
  >
    title
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#262626",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginTop": 8,
        },
      ]
    }
  >
    value
        
  </Text>
</View>
`;

exports[`Test InfoField render info field item without body 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#8C8C8C",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginTop": 8,
        },
      ]
    }
  >
    title
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#262626",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginTop": 8,
        },
      ]
    }
  >
    value
        
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#00979D",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      Chỉ đường
    </Text>
  </Text>
</View>
`;
